modules/apps/portlet-configuration/portlet-configuration-web/src/main/resources/META-INF/resources/edit_permissions.jsp

```diff
@@ -132,24 +131,70 @@ if (Validator.isNotNull(portletConfigurationPermissionsDisplayContext.getModelRe
 							String type = portletConfigurationPermissionsDisplayContext.getSelResourceDescription();
 
 							if (Validator.isNull(type)) {
-								type = ResourceActionsUtil.getModelResource(locale, resource.getName());
+								type = ResourceActionsUtil.getModelResource(locale, resourceName);
 							}
 
-							dataMessage = HtmlUtil.escapeAttribute(LanguageUtil.format(request, preselectedMsg, new Object[] {role.getTitle(locale), _getActionLabel(request, resource.getName(), action), type, HtmlUtil.escape(portletConfigurationPermissionsDisplayContext.getGroupDescriptiveName())}, false));
+							dataMessage = HtmlUtil.escapeAttribute(LanguageUtil.format(request, preselectedMsg, new Object[] {role.getTitle(locale), _getActionLabel(request, resourceName, action), type, HtmlUtil.escape(portletConfigurationPermissionsDisplayContext.getGroupDescriptiveName())}, false));
+
+							if (GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-87806"))) {
+								disabled = true;
+							}
 						}
 
 						String actionSeparator = Validator.isNotNull(preselectedMsg) ? ActionUtil.PRESELECTED : ActionUtil.ACTION;
 					%>
 
 						<liferay-ui:search-container-column-text
 							cssClass="table-column-text-center"
-							name="<%= _getActionLabel(request, resource.getName(), action) %>"
+							name="<%= _getActionLabel(request, resourceName, action) %>"
 						>
 							<c:if test="<%= disabled && checked %>">
 								<input name="<%= liferayPortletResponse.getNamespace() + role.getRoleId() + actionSeparator + action %>" type="hidden" value="<%= true %>" />
 							</c:if>
 
-							<input <%= checked ? "checked" : StringPool.BLANK %> class="<%= Validator.isNotNull(preselectedMsg) ? "lfr-checkbox-preselected lfr-portal-tooltip" : StringPool.BLANK %>" title="<%= dataMessage %>" <%= disabled ? "disabled" : StringPool.BLANK %> id="<%= FriendlyURLNormalizerUtil.normalize(role.getName()) + actionSeparator + action %>" name="<%= liferayPortletResponse.getNamespace() + role.getRoleId() + actionSeparator + action %>" onclick="<%= Validator.isNotNull(preselectedMsg) ? "return false;" : StringPool.BLANK %>" type="checkbox" />
+							<c:choose>
+								<c:when test='<%= GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-87806")) %>'>
+									<div>
+										<div class="custom-checkbox custom-control custom-control-inline">
+											<label>
+												<input
+													<%= (checked || indeterminate) ? "checked" : StringPool.BLANK %>
+													<%= disabled ? "disabled" : StringPool.BLANK %>
+													<%= indeterminate ? "value=\"indeterminate\"" : StringPool.BLANK %>
+													class="custom-control-input"
+													name="<%= liferayPortletResponse.getNamespace() + role.getRoleId() + actionSeparator + action %>"
+													type="checkbox"
+												/><span class="custom-control-label"></span
+											>
+											</label>
```

Comment:
Can we simplify this and remove the `<label>` and the empty `<span class="custom-control-label"></span>`? Just leave the input, it will the overrided by the react component.

Reply:
I don't think is a good idea, in mac, the native checkbook looks almost like Lexicon checkboxes but not in windows or Linux.

We use the `span.custom-control-label` to paint it the lexcion checbox.
![image](https://user-images.githubusercontent.com/67901/178236566-ea2edcea-55ad-45f8-86f3-f2b42b01ceff.png)

On the permissions modal, we have a huge matrix of checkboxes.

Reply:
ok, got it, thanks!